UpTraceCloudFunctions/acs/index.js

```diff
@@ -97,14 +100,20 @@ function authenticateToken(req) {
   return valid;
 }
 
+
+
+
 exports.acs = async (req, res) => {
   try {
     await injectSecrects();
     
     if(authenticateToken(req)) {
       //Case 1: JWT Valid
       console.log("Valid JWT")
-      res.redirect('https://attempt2-302520.ue.r.appspot.com/');
+      let functionParam = req.query.function
+      let helloController = new controller.HelloController(functionParam)
+      res.send(helloController.data);
```

Comment:
Ideally you should be able to do `require('controllers/HelloController.js')(req, res)` here. Also, the controller should not be responsible for determining which function it is. That would be the job of the api gateway.